### PR TITLE
oci: Handle timeouts correctly for probes

### PR DESCRIPTION
**Pull request by mrunalp**
We shouldn't return a custom error when an exec sync command
times out. Instead, we should return a non-zero exit code.

When the prober code in kubernetes sees a custom error
it goes into Unknown status and doesn't restart containers
on timed out probes as expected.

Signed-off-by: Mrunal Patel <mrunalp@gmail.com>

### CODE REVIEW

File: internal/oci/runtime_oci.go [hunk 1]
```diff
@@ -452,6 +452,21 @@ func (r *runtimeOCI) ExecSyncContainer(c *Container, command []string, timeout i
 
 	logrus.Debugf("Received container exit code: %v, message: %s", ec.ExitCode, ec.Message)
 
+	// This has to be in sync with the message returned by conmon
+	timedOutMessage := "command timed out"
```
Comment (rhatdan) [inline internal/oci/runtime_oci.go]: Can we get this message from a vendored conmon?

**Comment by mrunalp** [comment]
Yeah, I am going to open a PR to update conmon for master. We already did that for 1.16.

**Comment by haircommander** [comment]
taking this over [here](https://github.com/cri-o/cri-o/pull/3231)

**Pull request closed without merging**